routes/dao_coin_exchange.go

```diff
@@ -296,19 +328,85 @@ func buildDAOCoinLimitOrderResponse(
 
 		BuyingDAOCoinCreatorPublicKeyBase58Check:  buyingCoinPublicKeyBase58Check,
 		SellingDAOCoinCreatorPublicKeyBase58Check: sellingCoinPublicKeyBase58Check,
-		ExchangeRateCoinsToSellPerCoinToBuy:       exchangeRate,
-		QuantityToFill:                            quantityToFill,
+
+		Price:    price,
+		Quantity: quantity,
+
+		ExchangeRateCoinsToSellPerCoinToBuy: exchangeRate,
+		QuantityToFill:                      quantityToFill,
 
 		OperationType: operationTypeString,
 
 		OrderID: order.OrderID.String(),
 	}, nil
 }
 
-// CalculateScaledExchangeRate given a buying coin, selling coin, and a coin-level float exchange rate, this calculates
+///////////////////////////////////////////////////////////////////////////////////
+// Helper functions to calculate price and exchange rates for DAO coin limit orders
+///////////////////////////////////////////////////////////////////////////////////
+
+// CalculateScaledExchangeRateFromPriceString calculates a scaled ExchangeRateCoinsToSellPerCoinsToBuy given a decimal
+// price string (ex: "1.23456") that represents an exchange rate between the two coins where the denominator is
+// determined by the operation type.
+func CalculateScaledExchangeRateFromPriceString(
+	buyingCoinPublicKeyBase58Check string,
+	sellingCoinPublicKeyBase58Check string,
+	price string,
+	operationType lib.DAOCoinLimitOrderOperationType,
+) (*uint256.Int, error) {
+	rawScaledPrice, err := lib.CalculateScaledExchangeRateFromString(price)
+	if err != nil {
+		return nil, err
+	}
+	if rawScaledPrice.IsZero() {
+		return nil, errors.Errorf("The value %v is too small to produce a scaled exchange rate", price)
+	}
+
+	// Take the inverse of the exchange rate if the operation type is BID
+	rawScaledExchangeRate := rawScaledPrice
+	if operationType == lib.DAOCoinLimitOrderOperationTypeASK {
+		// Here we intend to calculate 1e38/price which gives us an ExchangeRateCoinsToSellPerCoinToBuy that's scaled up
+		// by 1e38. However, we can't avoid precision loss for irrational numbers, so we need to round up the quotient.
+		// The rounding allows ASK orders with irrational ExchangeRateCoinsToSellPerCoinToBuy values to match as expected
+		// with BID orders created using the same original input price. The integer division maths that gets us the intended
+		// result for ceil(1e38/price) using integer math is as follows:
+		//   ((1e38 + price) * 1e38) - 1) / (divisor * 1e38);
+		oneE72 := big.NewInt(0).Mul(lib.OneE38.ToBig(), lib.OneE38.ToBig())
+		numerator := big.NewInt(0).Add(oneE72, rawScaledExchangeRate.ToBig())
```

Comment:
Just trying to follow the math from your comment:
`((1e38 + price) * 1e38) - 1) / (divisor * 1e38);` becomes `((1e76 + (price * 1e38)) - 1) / (divisor * 1e38);`

is `rawScaledExchangeRate` already multiplied by 1e38? If not, I think the math is not quite right because this line would get us to `((1e76 + price) - 1) / (divisor * 1e38)`

Reply:
Exactly. We need to scale price up by 1e38 first in order to do the above division using integer math.